x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/SpatialPlugin.java

```diff
@@ -110,13 +122,49 @@ public void registerGeoShapeCentroidAggregator(ValuesSourceRegistry.Builder buil
             });
     }
 
-    public static void registerValueCountAggregator(ValuesSourceRegistry.Builder builder) {
+    public void registerGeoShapeGridAggregators(ValuesSourceRegistry.Builder builder) {
+        builder.register(GeoHashGridAggregationBuilder.NAME, GeoShapeValuesSourceType.instance(),
```

Comment:
Why other registers are static except this one? needs to be public?

Reply:
I made them all private. good catch.

The ones with license checks cannot be static because they need to call `getLicenseState()`

Reply:
Thanks for the explanation.